app/controllers/YourSelfAssessmentChargesController.scala

```diff
@@ -0,0 +1,120 @@
+import models.core.Nino
+import models.nextPayments.viewmodels.WYOClaimToAdjustViewModel
+import play.api.Logger
+import play.api.i18n.{I18nSupport, Messages}
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents, Result}
+import services.{ClaimToAdjustService, DateServiceInterface, WhatYouOweService}
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendController
+import views.html.WhatYouOwe
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class YourSelfAssessmentChargesController @Inject()(val authActions: AuthActions,
+                                     val whatYouOweService: WhatYouOweService,
+                                     val claimToAdjustService: ClaimToAdjustService,
+                                     val itvcErrorHandler: ItvcErrorHandler,
+                                     val itvcErrorHandlerAgent: AgentItvcErrorHandler,
+                                     val auditingService: AuditingService,
+                                     val dateService: DateServiceInterface,
+                                     whatYouOwe: WhatYouOwe
+                                    )(implicit val appConfig: FrontendAppConfig,
+                                      val mcc: MessagesControllerComponents,
+                                      val ec: ExecutionContext) extends FrontendController(mcc) with I18nSupport with FeatureSwitching{
+
+
+  def handleRequest(backUrl: String,
+                    itvcErrorHandler: ShowInternalServerError,
+                    isAgent: Boolean,
+                    origin: Option[String] = None)
+                   (implicit user: MtdItUser[_], hc: HeaderCarrier, ec: ExecutionContext, messages: Messages): Future[Result] = {
+
+    for {
+      whatYouOweChargesList <- whatYouOweService.getWhatYouOweChargesList(isEnabled(ReviewAndReconcilePoa), isEnabled(FilterCodedOutPoas))
+      ctaViewModel <- claimToAdjustViewModel(Nino(user.nino))
+    } yield {
+
+      auditingService.extendedAudit(WhatYouOweResponseAuditModel(user, whatYouOweChargesList, dateService))
+
+      val hasOverdueCharges: Boolean = whatYouOweChargesList.chargesList.exists(_.isOverdue()(dateService))
+      val hasAccruingInterestReviewAndReconcileCharges: Boolean = whatYouOweChargesList.chargesList.exists(_.isNotPaidAndNotOverduePoaReconciliationDebit()(dateService))
+      Ok(whatYouOwe(
+        currentDate = dateService.getCurrentDate,
+        hasOverdueOrAccruingInterestCharges = hasOverdueCharges || hasAccruingInterestReviewAndReconcileCharges,
+        whatYouOweChargesList = whatYouOweChargesList, hasLpiWithDunningLock = whatYouOweChargesList.hasLpiWithDunningLock,
+        currentTaxYear = dateService.getCurrentTaxYearEnd, backUrl = backUrl, utr = user.saUtr,
+        dunningLock = whatYouOweChargesList.hasDunningLock,
+        reviewAndReconcileEnabled = isEnabled(ReviewAndReconcilePoa),
+        isAgent = isAgent,
```

Comment:
same here, I am not sure if this should be part of tech debt or not?
but worth to rely on what we have in `user: MtdItUser[_]` when extracting `isAgent` status